src/main/java/org/sagebionetworks/bridge/redis/RedisLock.java

```diff
@@ -0,0 +1,57 @@
+package org.sagebionetworks.bridge.redis;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.UUID;
+
+import org.sagebionetworks.bridge.lock.Lock;
+import org.sagebionetworks.bridge.lock.LockNotAvailableException;
+
+public class RedisLock implements Lock {
+
+    private final JedisOps ops;
+
+    public RedisLock(JedisOps ops) {
+        checkNotNull(ops);
+        this.ops = ops;
+    }
+
+    @Override
+    public String acquireLock(final String key, final int expireInSeconds) {
+        checkNotNull(key);
+        checkArgument(expireInSeconds > 0);
+        final String lock = UUID.randomUUID().toString();
+        final Long result = ops.setnx(key, lock);
+        if (result != 1L) {
+            // When fail to obtain the lock
+            // Force to expire the key if not already set to expire
+            final Long expire = ops.ttl(key);
+            if (expire < 0L || expire > expireInSeconds) {
+                expire(key, expireInSeconds);
+            }
```

Comment:
Good catch.  I think I should explicitly check against -1, which indicates no associated expiration for Redis 2.8 and later.  The intention is to avoid starvation.  Imagine if the lock has already been acquired somewhere else but without setting the expiration, e.g. when the app has crashed after line 25 `ops.setnx(key, lock)` but before line 35 `expire(key, expireInSeconds)`.  The lock will never be available.